src/backend/distributed/commands/table.c

```diff
@@ -162,31 +163,61 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 
 
 /*
- * PostprocessCreateTableStmt takes CreateStmt object as a parameter and errors
- * out if it creates a table with a foreign key that references to a citus local
- * table.
+ * PostprocessCreateTableStmt takes CreateStmt object as a parameter and
+ * processes foreign keys on relation via PostprocessCreateTableStmtForeignKeys
+ * function.
  *
  * This function also processes CREATE TABLE ... PARTITION OF statements via
  * PostprocessCreateTableStmtPartitionOf function.
  */
 void
 PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
+{
+	PostprocessCreateTableStmtForeignKeys(createStatement);
+
+	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+	{
+		/* process CREATE TABLE ... PARTITION OF command */
+		PostprocessCreateTableStmtPartitionOf(createStatement, queryString);
+	}
+}
+
+
+/*
+ * PostprocessCreateTableStmtForeignKeys drops ands re-defines foreign keys
+ * defined by given CREATE TABLE command if command defined any foreign to
+ * reference or citus local tables.
+ */
+static void
+PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 {
 	/*
 	 * Relation must exist and it is already locked as standard process utility
 	 * is already executed.
 	 */
 	bool missingOk = false;
 	Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
-	if (HasForeignKeyToCitusLocalTable(relationId))
+
+	if (!(TableReferenced(relationId) || TableReferencing(relationId)))
 	{
-		ErrorOutForFKeyBetweenPostgresAndCitusLocalTable(relationId);
+		return;
 	}
 
-	if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+	List *nonDistTableForeignKeyIdList =
+		GetForeignKeyOids(relationId, INCLUDE_REFERENCING_CONSTRAINTS |
+						  INCLUDE_CITUS_LOCAL_TABLES |
+						  INCLUDE_REFERENCE_TABLES);
+	bool hasForeignKeyToNonDistTable = list_length(nonDistTableForeignKeyIdList) != 0;
+	if (hasForeignKeyToNonDistTable)
```

Comment:
"NonDist" is a little confusing here. 

maybe `hasForeignKeyToCitusTableOnCoordinator`